### DISCUSSION TITLE
Pico H with LAN support (Lan8720)

### DISCUSSION THREAD

**Discussion by Erkman84** [micropython/micropython#15829] [RP2040 / Pico]
Hi there,
I've tried to build micropython on a raspbian, but libsso 1.1 failed (there were some tries to fix but did not work). This is not my question yet.

What I've would like to achive:
get a ufw file for the causal Pico H (not W) a firmware where the Network.LAN module (and nice to have: usocket, urequests, mip) is implemented. This is the one which I needed. However the Pico W's firmware does not have the LAN interface as default in the Network library (and you cannot add this), so that's also not a way to go.
Can anybody help me please?

**Comment by andrewleech** [comment]
I'm not sure if anyone has written a driver for rp2xxx that supports the Lan8720 controller, however I do know the wiznet chips work , eg.

https://micropython.org/download/W5100S_EVB_PICO/
https://micropython.org/download/W5500_EVB_PICO/

The wiznet is also available as a hat that the same firmware works for: https://docs.wiznet.io/Product/Open-Source-Hardware/wiznet_ethernet_hat

**Reply by Erkman84** [reply]
Hi, the wiznet is for another chip, I saw it earlier, but as it says, it works with only with wiznet. Does this driver also work with the LAN8720? There is no documentation for it....

**Reply by andrewleech** [reply]
No, as I started with I haven't heard of anyone writing a driver for the lanxxxx brand of controllers on rp2. 

Oh wait I just remembered the Lan8720 and similar are just the PHY, it requires a separate ethernet MAC. When this chip is used with eg. stm32 or esp32 it's wired to the ethernet MAC peripheral in these chips. 

By comparison the wiznet chips are a full-stack controller that doesn't require anything from the MCU other than spi for communication.

rp2 does not have a built in ethernet MAC, but I see Sandeep Mistry has written a software mac for them using PIO and DMA https://www.raspberrypi.com/news/how-to-add-ethernet-to-raspberry-pi-pico/

In theory LWIP, PIO and DMA is all supported in micropython now, but I don't see any suggestion that anyone has ported this combination in micropython. I'm really not sure how much effort that would take sorry.

**Comment by bidrohini** [comment]
You need to add a Wiznet W5500.

**Reply by Erkman84** [reply]
Please read previous conversation, as you can see, wizznet driver is not working, it is a SPI interface, and LAN8720 is NOT a SPI interface based communication device, so the same driver will not work.
